src/easylink/graph_components.py

```diff
@@ -41,20 +42,48 @@ class InputSlot:
     env_var: str | None
     """The environment variable that is used to pass a list  of data filepaths to 
     an ``Implementation``."""
-    validator: Callable[[str], None]
+    validator: Callable[[str], None] = field(compare=False)
     """A function that validates the input data being passed into the pipeline via 
     this ``InputSlot``. If the data is invalid, the function should raise an exception 
     with a descriptive error message which will then be reported to the user.
     **Note that the function *must* be defined in the** :mod:`easylink.utilities.validation_utils` 
     **module!**"""
-    splitter: Callable[[list[str], str, Any], None] | None = None
+    splitter: Callable[[list[str], str, Any], None] | None = field(
+        default=None, compare=False
+    )
     """A function that splits the incoming data to this ``InputSlot`` into smaller
     pieces. The primary purpose of this functionality is to run sections of the 
     pipeline in an embarrassingly parallel manner. **Note that the function *must* 
     be defined in the **:mod:`easylink.utilities.splitter_utils`** module!**"""
 
+    def __eq__(self, other: Any) -> bool | NotImplementedType:
```

Comment:
also, we don't seem to have any instances currently of actually checking ==, but I figured as long as I'm here I'd add it with the comparable hashing logic